From 909b254eb818f63ba0e541abbbe829d923c19234 Mon Sep 17 00:00:00 2001
From: Daniel Martin <consume.noise@gmail.com>
Date: Fri, 2 Oct 2015 10:07:15 +0200
Subject: [PATCH 2/4] HID: multitouch: Add MT_QUIRK_NOT_SEEN_MEANS_UP to
 MT_CLS_WIN_8

The firmware found in the touch screen of an SP3 is buggy and may miss
to send lift off reports for contacts. Try to work around that issue by
using MT_QUIRK_NOT_SEEN_MEANS_UP.

Signed-off-by: Daniel Martin <consume.noise@gmail.com>
---
diff -Naur linux-4.13-orig/drivers/hid/hid-multitouch.c linux-4.13/drivers/hid/hid-multitouch.c
--- linux-4.13-orig/drivers/hid/hid-multitouch.c	2017-09-03 15:56:17.000000000 -0500
+++ linux-4.13/drivers/hid/hid-multitouch.c	2017-09-03 21:42:00.082579541 -0500
@@ -223,6 +223,7 @@
 			MT_QUIRK_IGNORE_DUPLICATES |
 			MT_QUIRK_HOVERING |
 			MT_QUIRK_CONTACT_CNT_ACCURATE |
+			MT_QUIRK_NOT_SEEN_MEANS_UP |
 			MT_QUIRK_STICKY_FINGERS },
 	{ .name = MT_CLS_EXPORT_ALL_INPUTS,
 		.quirks = MT_QUIRK_ALWAYS_VALID |
@@ -1254,7 +1255,8 @@
 		return -ENOMEM;
 	}
 
-	if (id->vendor == HID_ANY_ID && id->product == HID_ANY_ID)
+	if (id->vendor == HID_ANY_ID && id->product == HID_ANY_ID &&
+		id->group != HID_GROUP_MULTITOUCH_WIN_8)
 		td->serial_maybe = true;
 
 	/*s
-- 
2.1.4


